app/src/androidTest/java/com/droidhats/campuscompass/views/MapFragmentTest.kt

```diff
@@ -54,12 +64,211 @@ class MapFragmentTest {
         }
     }
 
-    //Checking if the search bar exists
+
     @Test
     fun test_SearchBar() {
+        //Checks if the search bar with text "search" exists
+        val searchTextView = onView(
+            allOf(
+                withId(R.id.mt_placeholder), withText("Search"), isDisplayed()
+            )
+        )
+        searchTextView.check(matches(withText("Search")))
+
+        //Checks if magnifying glass button is displayed
+        val magnifyingGlassView = onView(
+            allOf(
+                withId(R.id.mt_search), isDisplayed()
+            )
+        )
+        magnifyingGlassView.check(matches(isDisplayed()))
+
+        //Checks if Search bar can be clicked
+        val containerSearchBar = onView(
+            allOf(
+                withId(R.id.mapFragSearchBar), isDisplayed()
+            )
+        )
+        containerSearchBar.perform(click())
+
+        //Checks if text "Search" is displayed even after search bar was clicked
+        val searchText = onView(
+            allOf(
+                withId(R.id.search_src_text), withHint("Search"), isDisplayed()
+            )
+        )
+        searchText.check(matches(withHint("Search")))
+
+        //Checks if info text that prompts to enter the text in the text field is displayed
+        val infoTextView = onView(
+            allOf(
+                withId(R.id.search_info), withText("Enter Street, Address, Concordia Classroom..."),
+                isDisplayed()
+            )
+        )
+        infoTextView.check(matches(withText("Enter Street, Address, Concordia Classroom...")))
```

Comment:
As mentioned earlier, avoid declaring variable for onView method specially since it is not being reused.